packages/react-native-reanimated/src/layoutReanimation/animationBuilder/commonTypes.ts

```diff
@@ -80,7 +80,7 @@ export type LayoutAnimationStartFunction = (
   tag: number,
   type: LayoutAnimationType,
   yogaValues: Partial<SharedTransitionAnimationsValues>,
-  config: LayoutAnimationFunction
+  config: (arg: Partial<SharedTransitionAnimationsValues>) => LayoutAnimation
```

Comment:
Why do we need to change this type? 🤔

Reply:
We have defined and exported type `LayoutAnimationStartFunction`, which was expected to be used to type function `createLayoutAnimationManager`. However this type got suddenly disconnected from the function and the type wasn't used anywhere across our codebase. Here I connect them back: https://github.com/software-mansion/react-native-reanimated/pull/6235#discussion_r1678966967
For this reason the type and the function went out of sync.